src/service/routing/directions.clj

```diff
@@ -65,21 +70,25 @@
    Example:
    (direction graph :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]"
   [graph & parameters]
-  (let [{:keys [coordinates steps]} parameters
-        start     (brute-nearest graph (first coordinates))
-        dst       (brute-nearest graph (last coordinates))
-        traversal (alg/dijkstra graph :value-by length
-                                      :start-from #{(key start)})
-        trace     (reduce (fn [_ trace] (when (= (key trace) (key dst)) (reduced trace)))
-                          nil
-                          traversal)]
-    (if (nil? trace)
-      {:code "NoRoute"}
-      {:code "Ok"
-       :waypoints (map (fn [point] {:name "wonderland"
-                                    :location [(:lon point) (:lat point)]})
-                       coordinates)
-       :routes [(route graph trace)]})))
+  (if-not (validate-radiuses parameters)
```

Comment:
> Either the spec should support it or it should be done as part of directions consistency checks

Actually both. See [Using Spec for validation](https://clojure.org/guides/spec). It is quite normal (specially for server applications) to need to have runtime validation of input. It just happened that `compojure-api` made it really easy for us to avoid having to write that code manually but normally you would need to write that code manually.

So what I had in mind was:
- [x] decouple spec from route def
- [ ] create a spec on runtime (but dont register it) based on the coordinates element
- [ ] validate that the radiouses also conform to the newly created spec

So something like this should work:
```clojure
; in spec ns
(s/def ::radious (s/or #(= "undefined" %) (s/and number? #(>= % 0 )))
(s/def ::radiouses (s/coll-of ::radious))

;; in directions ns
(:require [service.routing.specs :as specs]) ;; need to import it

;; in the directions functions. Create a spec at runtime without registering it and call valid? to see
;; if the radiouses conforms to the newly created spec
(if-not (s/valid? (s/and :specs/radiouses #(= (count coordinates) (count %)) parameters)
 ; ... report error here
```

Of course for that to work you would need to move the destructuring up to the function arguments not in the let as it currently is ;)